provider/confluence/api.ts

```diff
@@ -0,0 +1,70 @@
+import type { Settings } from './index.ts'
+
+const authHeaders = (settings: Settings) => ({
+    Authorization: `Basic ${Buffer.from(`${settings.email}:${settings.apiToken}`).toString('base64')}`,
+})
+
+const endpoint = (settings: Settings) =>
+    'https://' + settings.host + (settings.port ? `:${settings.port}` : '')
```

Comment:
minor: what do you think of excluding the settings.port and users just including port in the host field. Host is a reasonable name for this, eg `new URL('https://example.com:10810').host` will be `example.com:10810`